pkg/cloud/option.go

```diff
@@ -0,0 +1,27 @@
+package cloud
+
+// Option are optional parameters to the generated methods.
+type Option struct {
+	projectID string
+}
+
+func (co *Option) mergeInto(other *Option) {
+	if co.projectID != "" {
+		other.projectID = co.projectID
+	}
+}
+
+func mergeOptions(options []Option) Option {
+	var ret Option
+	for _, opt := range options {
+		opt.mergeInto(&ret)
+	}
```

Comment:
this will be totally dependen on the order of the options passed, no?
basically this is "use the projectID from the last option with this field set"

Reply:
Fixed -- changed Option to an interface with the merge into allOptions struct so this is clear.